### ISSUE TITLE
<!-- -*- mode: text --> in markdown document or wiki page breaks Web/HTML rendering

### ISSUE THREAD

**Issue by sideshowbarker** [isaacs/github#904]
Adding `<!-- -*- mode: text -->` to a GitHub markdown document or GitHub Wiki page seems to break GitHub’s Web/HTML rendering of the document/page.

I’m not an Emacs used but my understanding is that it’s not uncommon for people to use Emacs to locally edit their GitHub markdown documents and source for GitHub Wiki pages. And for those users, it’s convenient to include that `<!-- -*- mode: text -->` thing to set the Emacs local variable that put Emacs into the right mode.

**Comment by xfq** [comment]
Hmm, although I can reproduce this issue, as an Emacs user & committer, I don't understand why one would add the `mode: text` file variable to a Markdown file.

Although Emacs doesn't come with Markdown support, Emacs users usually install and use [markdown-mode](http://jblevins.org/git/markdown-mode.git/) (instead of [text-mode](https://www.gnu.org/software/emacs/manual/html_node/emacs/Text-Mode.html#Text-Mode)) for their Markdown editing needs.

Since I might be missing something, can you elaborate why you added `mode: text` to [this document](https://github.com/validator/validator/wiki/Microsyntax-descriptions/_compare/7d272874ff5a44d09094b0c832db33b7a69d5422...384929c60b71ce32c7673ccc6d30246e973d36cb), @sideshowbarker?

**Comment by sideshowbarker** [comment]
Since Emacs doesn’t have built-in markdown support, I had assumed (wrongly I guess…) that the closest thing was `text` mode, because otherwise Emacs just uses `Fundamental` mode (I think?) and I also assumed (maybe wrongly again…) that `Fundamental` mode was to be avoided if something more specific mode is appropriate.

But if setting `mode: text` is the wrong thing to do then I guess the resulting GitHub behavior is a non-issue—because it would seem like the right fix to avoid the problem altogether would then be to just not set `mode: text` for markdown files.

**Comment by xfq** [comment]
> Since Emacs doesn’t have built-in markdown support, I had assumed (wrongly I guess…) that the closest thing was text mode,

You're right. Actually, `markdown-mode` is [derived](http://jblevins.org/git/markdown-mode.git/tree/markdown-mode.el?id=0f88075c9091b4120fac96aaf091ec647d2ea555#n6684) from `text-mode`.

> because otherwise Emacs just uses Fundamental mode (I think?) and I also assumed (maybe wrongly again…) that Fundamental mode was to be avoided if something more specific mode is appropriate.

Right, too. However, adding `mode: text` to Markdown files will make users who have `markdown-mode` use plain `text-mode`, losing all Markdown-specific syntactic highlighting, key bindings, and other customization, which is not user-friendly.

**Comment by sideshowbarker** [comment]
> adding `mode: text` to Markdown files will make users who installed `markdown-mode` use plain `text-mode`, losing all Markdown-specific syntactic highlighting, key bindings, and other customization, which is not user-friendly

OK, yeah, given that it pretty clearly seems like having files or wiki pages in any shared repo with `mode: text` set on them is bad and should be avoided.

That said, I can imagine a use-case where somebody personally doesn’t like markdown mode (for whatever reason) and intentionally doesn’t install and use it, but instead wants to use `text` mode, and so has the mode set that way in the markdown files in their repo (which may just be a personal repo not intended for shared use).

But that use-case seems a bit far-fetched, not something that ought to optimized for in any way.
